async function fetchDataWithErrorHandling() {
    try {
        const response = await fetch('https://swapi.dev/api/people/1');
        const characterData = await response.json();

        if(!response.ok) {
            throw new Error('Something went wrong');
        }

        console.log(characterData);
    }
    catch(err) {
        console.error('Errorrrrrrr', err.message);
    }
}

fetchDataWithErrorHandling();
